great_expectations/profile/json_schema_profiler.py

```diff
@@ -0,0 +1,253 @@
+import abc
+from enum import Enum
+from typing import Any, Dict, Optional, Union
+
+from great_expectations.core import (
+    ExpectationConfiguration,
+    ExpectationKwargs,
+    ExpectationSuite,
+)
+
+
+class JsonSchemaTypes(Enum):
+    STRING = "string"
+    INTEGER = "integer"
+    NUMBER = "number"
+    ARRAY = "array"
+    NULL = "null"
+    BOOLEAN = "boolean"
+    OBJECT = "object"
+    ENUM = "enum"
+
+
+class NonDataProfiler(object, metaclass=abc.ABCMeta):
```

Comment:
Both `DataAssetProfiler` and `JsonSchemaProfiler` should inherit from `Profiler`.

This "NonDataProfiler" still has parts of the JsonSchemaProfiler in it (e.g. param name for profile and validate logic). Please update to be a generic base class called `Profiler` that defines the Profiler API, and then collapse all of the JsonSchema specific logic into the JsonSchemaProfiler.

Reply:
Done. Great call.